File: LC 2735.cpp
class Solution {
public:
    long long minCost(vector<int>& nums, int x) {
        long long ans = 0;
        vector<int>a(nums);
        for (int x : nums) ans += x;
        long long s = 0;
        for (int i = 0; i < nums.size(); i++) {
            s = 0;
            for (int j = 0; j < nums.size(); j++) {
                a[j] = min(a[j], nums[(j-i+nums.size())%nums.size()]);
                s += a[j];
            }
            ans = min(ans, (long long)x * i + s);
        }
        return ans;
    }
};

File: 3000/LC 2735.go
func minCost(nums []int, x int) (ans int64) {
    for _, x := range nums {
        ans += int64(x)
    }

    n := len(nums)
    a := append([]int(nil), nums...)
    for i := range nums {
        var s int64
        for j := range nums {
            a[j] = min(a[j], nums[(j - i + n) % n])
            s += int64(a[j])
        }
        ans = min(ans, s + int64(x) * int64(i))
    }

    return 
}